chia/server/node_discovery.py

```diff
@@ -80,6 +82,20 @@ def __init__(
         if default_port is None and selected_network in NETWORK_ID_DEFAULT_PORTS:
             self.default_port = NETWORK_ID_DEFAULT_PORTS[selected_network]
 
+    async def migrate_address_manager_if_necessary(self) -> None:
+        if self.legacy_peer_db_path is None or not self.legacy_peer_db_path.exists() or self.peers_file_path.exists():
+            return
+        try:
+            self.log.info(f"Migrating legacy peer database from {self.legacy_peer_db_path}")
+            # Attempt to create an AddressManager from the legacy peer database
+            address_manager: Optional[AddressManager] = await AddressManagerSQLiteStore.create(self.legacy_peer_db_path)
+            if address_manager is not None:
+                self.log.info(f"Writing migrated peer data to {self.peers_file_path}")
+                # Write the AddressManager data to the new peers file
+                await AddressManagerStore.serialize(address_manager, self.peers_file_path)
```

Comment:
if we get here without an error, shouldn't `self.legacy_peer_db_path` be set to `None`? I suppose this is only called on `start()` so maybe not. However, we don't delete (or move aside) the old DB. How do we record that we've already migrated, to avoid doing it again next time we restart (and clobber our new peer list with an old one)?

Reply:
I'll add a `migrated` instance variable to track this within the same run to make things a little tidier. Though, once migrated, the new file exists, so we won't reattempt migration.

Reply:
I see, we should probably check for its existence first then (before the db)